demisto_sdk/commands/test_content/tests/timestamp_replacer_test.py

```diff
@@ -0,0 +1,233 @@
+import json
+import sys
+from collections import OrderedDict
+from unittest.mock import MagicMock, mock_open
+
+import mitmproxy
+import pytest
+from demisto_sdk.commands.test_content.timestamp_replacer import \
+    TimestampReplacer
+from mitmproxy.http import HTTPFlow, HTTPRequest
+from mitmproxy.net.http import Headers
+from munch import Munch
+
+
+@pytest.fixture()
+def flow():
+    request = HTTPRequest(first_line_format='first_line',
+                          host=b'localhost',
+                          path=b'/test/',
+                          http_version=b'1.1',
+                          port=1234,
+                          method=b'',
+                          scheme=b'',
+                          headers=Headers([(b"Host", b"example.com")]),
+                          content=None,
+                          timestamp_start=111.1)
+    flow = HTTPFlow(client_conn=MagicMock(),
+                    server_conn=MagicMock())
+    flow.request = request
+    return flow
+
+
+class TestTimeStampReplacer:
+    DEFAULT_OPTIONS_MAPPING = {
+        'detect_timestamps': False,
+        'keys_filepath': 'problematic_keys.json',
+        'script_mode': 'playback',
+        'debug': False
+    }
+
+    @classmethod
+    def setup_class(cls):
+        mitmproxy.ctx.options = Munch(**cls.DEFAULT_OPTIONS_MAPPING)
```

Comment:
How come this works? Does `pytest` automatically invoke classmethods defined on classes with the `Test` prefix on the instantiated instance before running all the methods?

Reply:
Yup.
Plus it's my magic

Reply:
https://github.com/szabgab/slides/blob/main/python/examples/pytest/test_class.py

Reply:
Cool. I wasn't aware of this feature. Good to know.